lib/src/merge.rs

```diff
@@ -816,6 +1034,55 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_update_from_simplified() {
+        // 1-way merge
+        assert_eq!(
+            c(&[], &[0]).update_from_simplified(c(&[], &[1])),
+            c(&[], &[1])
+        );
+        // 3-way merge
+        assert_eq!(
+            c(&[0], &[0, 0]).update_from_simplified(c(&[], &[1])),
+            c(&[0], &[0, 1])
+        );
+        assert_eq!(
+            c(&[0], &[1, 0]).update_from_simplified(c(&[], &[2])),
+            c(&[0], &[2, 0])
+        );
+        assert_eq!(
+            c(&[0], &[1, 2]).update_from_simplified(c(&[1], &[2, 3])),
+            c(&[1], &[2, 3])
+        );
+        // 5-way merge
+        assert_eq!(
+            c(&[0, 0], &[0, 0, 0]).update_from_simplified(c(&[], &[1])),
+            c(&[0, 0], &[0, 0, 1])
+        );
+        assert_eq!(
+            c(&[0, 1], &[0, 0, 0]).update_from_simplified(c(&[3], &[2, 1])),
+            c(&[0, 3], &[0, 2, 1])
+        );
+        assert_eq!(
+            c(&[1, 0], &[0, 0, 0]).update_from_simplified(c(&[3], &[2, 1])),
+            c(&[3, 0], &[0, 1, 2])
+        );
+        assert_eq!(
+            c(&[0, 1], &[2, 3, 4]).update_from_simplified(c(&[1, 2], &[3, 4, 5])),
+            c(&[1, 2], &[3, 4, 5])
+        );
+        assert_eq!(
```

Comment:
If it's not a fuss, I'd put a newline before this to make it clearer that this assertion is context for the next one.

If it is a fuss, I'll create a PR to do that or something here.

Thank you very much for adding the test and all the fixes!